l1-contracts/contracts/governance/PermanentRestriction.sol

```diff
@@ -219,6 +219,13 @@ contract PermanentRestriction is Restriction, IPermanentRestriction, Ownable2Ste
         // - Query the Bridgehub for the Hyperchain with the given `chainId`.
         // - We compare the corresponding addresses
 
+        // Note, that we do not use an explicit call here to ensure that the function does not panic in case of
+        // incorrect `_chain` address.
+        (bool success, bytes memory data) = _chain.staticcall(abi.encodeWithSelector(IGetters.getChainId.selector));
```

Comment:
encodeWithCall is better because it is compiler time checks the interface. This was OZ issue reported so better to fix it here too